Roll Dart to 4cf2d3990bac1f1df2f524899e1560a3683f3133

aartbik opened the pull request:
Included CLs:

4cf2d3990b [VM runtime] Store Smi instance type in type test cache when using the interpreter.
a6b1a99f4c Make mixins abstract by default
ab305f2f26 Recover from duplicated declarations
4e32a356e0 Ignore synthetic elements created by the parser
fc04f56430 [infra] Remove deleted Dart 1 builders (app/precomp) from the test matrix
8477dc2131 Update status after 570fd5a7884
1652850381 Compute static types during impact computation
b4341f02db Remove unneeded method rti need
570fd5a788 Error on re-export, export and import of libraries with same name.
4b312b2b8c Support multiple superclasses in the mixin `on` clause.
8debb4e9db [fuchsia] Add zx_vmo_replace_as_executable calls
d34337aa2d Ignore noSuchMethod() for the purpose of searching concrete member implementations.
b6284b41e3 Revert "[VM] Dart_Initialize no longer crashes after Dart_Cleanup"
94481adbd4 [vm] Only include debugger/profiler slots in RawCode in product mode.
519ee905f9 [VM] Dart_Initialize no longer crashes after Dart_Cleanup
53497549ab [vm] Specialize 'new List()' on kernel AST
afd4cc9d1a [vm/tfa] Infer concrete class of int literals in TFA
a4d48f49e7 [vm/tfa] Infer concrete class of string literals in TFA
9106228a55 [vm, dwarf] Set default_is_stmt to true for compatability with dsymutil.
6ce0dc0006 dartfix stub implementation
e4eb2437b8 Look into superclassConstraints while searching for a member in interfaces.
95d37b0921 Revert "Implement inheritance/override checks from the spec."
355c73d4ee [VM] Remove support for option --no-preview-dart-2
95b8a19a20 Mixin declarations don't have supertype, fix isMoreSpecificThan().
e9586a5d14 [vm] Update workaround for Android Kernel bug (b089d4f).
ef7d144bc7 Add regression test for issue 34392.
56f6c52d58 large_class_declaration_test is slow now.
0ac3b8fff9 [fuchsia] Add missing vm:entry-point annotation
19e844ed5b [dart/fuzzing] Use current directory as SDK top (by default)
58e44c1400 Don't use ClassElementImpl for now in override checking.
62d154f6a3 [vm/bytecode] Record null-initialized fields in bytecode
70cc34df21 Custom format errors and exceptions to show traces
95831136b4 Fix NPE when accessing superclass
836a1d7a88 Implement inheritance/override checks from the spec.
d07039782a Remove redundant resolution
dba8b37d38 [vm] Benchmark for reading kernel bytecode

aartbik committed:
Roll Dart to 4cf2d3990bac1f1df2f524899e1560a3683f3133

Included CLs:

4cf2d3990b [VM runtime] Store Smi instance type in type test cache when using the interpreter.
a6b1a99f4c Make mixins abstract by default
ab305f2f26 Recover from duplicated declarations
4e32a356e0 Ignore synthetic elements created by the parser
fc04f56430 [infra] Remove deleted Dart 1 builders (app/precomp) from the test matrix
8477dc2131 Update status after 570fd5a7884
1652850381 Compute static types during impact computation
b4341f02db Remove unneeded method rti need
570fd5a788 Error on re-export, export and import of libraries with same name.
4b312b2b8c Support multiple superclasses in the mixin `on` clause.
8debb4e9db [fuchsia] Add zx_vmo_replace_as_executable calls
d34337aa2d Ignore noSuchMethod() for the purpose of searching concrete member implementations.
b6284b41e3 Revert "[VM] Dart_Initialize no longer crashes after Dart_Cleanup"
94481adbd4 [vm] Only include debugger/profiler slots in RawCode in product mode.
519ee905f9 [VM] Dart_Initialize no longer crashes after Dart_Cleanup
53497549ab [vm] Specialize 'new List()' on kernel AST
afd4cc9d1a [vm/tfa] Infer concrete class of int literals in TFA
a4d48f49e7 [vm/tfa] Infer concrete class of string literals in TFA
9106228a55 [vm, dwarf] Set default_is_stmt to true for compatability with dsymutil.
6ce0dc0006 dartfix stub implementation
e4eb2437b8 Look into superclassConstraints while searching for a member in interfaces.
95d37b0921 Revert "Implement inheritance/override checks from the spec."
355c73d4ee [VM] Remove support for option --no-preview-dart-2
95b8a19a20 Mixin declarations don't have supertype, fix isMoreSpecificThan().
e9586a5d14 [vm] Update workaround for Android Kernel bug (b089d4f).
ef7d144bc7 Add regression test for issue 34392.
56f6c52d58 large_class_declaration_test is slow now.
0ac3b8fff9 [fuchsia] Add missing vm:entry-point annotation
19e844ed5b [dart/fuzzing] Use current directory as SDK top (by default)
58e44c1400 Don't use ClassElementImpl for now in override checking.
62d154f6a3 [vm/bytecode] Record null-initialized fields in bytecode
70cc34df21 Custom format errors and exceptions to show traces
95831136b4 Fix NPE when accessing superclass
836a1d7a88 Implement inheritance/override checks from the spec.
d07039782a Remove redundant resolution
dba8b37d38 [vm] Benchmark for reading kernel bytecode

goderbauer reviewed (approved):
LGTM

goderbauer commented:
Looks like cirrus is having trouble downloading or unpacking the Dart SDK for this engine revision.

aartbik committed:
Move one revision later.

Rationale:
Relatively harmless change, but at least has
builds (force builds on the older revision
will take a while again).
